app/celery/letters_pdf_tasks.py

```diff
@@ -1,424 +0,0 @@
-    """
-    running_filesize = 0
-    list_of_files = []
-    for letter in letter_pdfs:
-        if letter['Key'].lower().endswith('.pdf') and letter_in_created_state(letter['Key']):
-            if (
-                running_filesize + letter['Size'] > current_app.config['MAX_LETTER_PDF_ZIP_FILESIZE']
-                or len(list_of_files) >= current_app.config['MAX_LETTER_PDF_COUNT_PER_ZIP']
-            ):
-                yield list_of_files
-                running_filesize = 0
-                list_of_files = []
-
-            running_filesize += letter['Size']
-            list_of_files.append(letter)
-
-    if list_of_files:
-        yield list_of_files
-
-
-def letter_in_created_state(filename):
-    # filename looks like '2018-01-13/NOTIFY.ABCDEF1234567890.D.2.C.C.20180113120000.PDF'
-    subfolder = filename.split('/')[0]
-    ref = get_reference_from_filename(filename)
-    notifications = dao_get_notifications_by_references([ref])
-    if notifications:
-        if notifications[0].status == NOTIFICATION_CREATED:
-            return True
-        current_app.logger.info(
-            'Collating letters for %s but notification with reference %s already in %s',
-            subfolder,
-            ref,
-            notifications[0].status,
-        )
-    return False
-
-
-@notify_celery.task(bind=True, name='process-virus-scan-passed', max_retries=15, default_retry_delay=300)
-def process_virus_scan_passed(
-    self,
-    filename,
-):
-    reference = get_reference_from_filename(filename)
-    notification = dao_get_notification_by_reference(reference)
-    current_app.logger.info('notification id %s Virus scan passed: %s', notification.id, filename)
-
-    is_test_key = notification.key_type == KEY_TYPE_TEST
-
-    scan_pdf_object = s3.get_s3_object(current_app.config['LETTERS_SCAN_BUCKET_NAME'], filename)
-    old_pdf = scan_pdf_object.get()['Body'].read()
-
-    try:
-        billable_units = get_page_count(old_pdf)
-    except PdfReadError as e:
-        current_app.logger.exception(e)
-        current_app.logger.error('Invalid PDF received for notification_id: %s', notification.id)
-        _move_invalid_letter_and_update_status(notification, filename, scan_pdf_object)
-        return
-
-    sanitise_response = _sanitise_precompiled_pdf(self, notification, old_pdf)
-    if not sanitise_response:
-        new_pdf = None
-    else:
-        sanitise_response = sanitise_response.json()
-        try:
-            new_pdf = base64.b64decode(sanitise_response['file'].encode())
-        except JSONDecodeError:
-            new_pdf = sanitise_response.content
-
-        redaction_failed_message = sanitise_response.get('redaction_failed_message')
-        if redaction_failed_message and not is_test_key:
-            current_app.logger.info(
-                '%s for notification id %s (%s)', redaction_failed_message, notification.id, filename
-            )
-            copy_redaction_failed_pdf(filename)
```

Comment:
`copy_redaction_failed_pdf` can be removed - it's only called here and in testing. `app.letters.utils.copy_redaction_failed_pdf.py`